plugins/package-managers/gradle-inspector/src/funTest/assets/projects/synthetic/gradle-expected-output-lib-without-repo.yml

```diff
@@ -17,57 +17,22 @@ project:
   homepage_url: ""
   scopes:
   - name: "compileClasspath"
-    dependencies:
-    - id: "Unknown:org.apache.commons:commons-text:1.1"
-      issues:
-      - timestamp: "1970-01-01T00:00:00Z"
-        source: "Gradle"
-        message: "Unresolved: ModuleVersionNotFoundException: Cannot resolve external\
-          \ dependency org.apache.commons:commons-text:1.1 because no repositories\
-          \ are defined."
-        severity: "ERROR"
+    dependencies: []
   - name: "runtimeClasspath"
-    dependencies:
-    - id: "Unknown:org.apache.commons:commons-text:1.1"
-      issues:
-      - timestamp: "1970-01-01T00:00:00Z"
-        source: "Gradle"
-        message: "Unresolved: ModuleVersionNotFoundException: Cannot resolve external\
-          \ dependency org.apache.commons:commons-text:1.1 because no repositories\
-          \ are defined."
-        severity: "ERROR"
+    dependencies: []
   - name: "testCompileClasspath"
-    dependencies:
-    - id: "Unknown:junit:junit:4.12"
-      issues:
-      - timestamp: "1970-01-01T00:00:00Z"
-        source: "Gradle"
-        message: "Unresolved: ModuleVersionNotFoundException: Cannot resolve external\
-          \ dependency junit:junit:4.12 because no repositories are defined."
-        severity: "ERROR"
-    - id: "Unknown:org.apache.commons:commons-text:1.1"
-      issues:
-      - timestamp: "1970-01-01T00:00:00Z"
-        source: "Gradle"
-        message: "Unresolved: ModuleVersionNotFoundException: Cannot resolve external\
-          \ dependency org.apache.commons:commons-text:1.1 because no repositories\
-          \ are defined."
-        severity: "ERROR"
+    dependencies: []
   - name: "testRuntimeClasspath"
-    dependencies:
-    - id: "Unknown:junit:junit:4.12"
-      issues:
-      - timestamp: "1970-01-01T00:00:00Z"
-        source: "Gradle"
-        message: "Unresolved: ModuleVersionNotFoundException: Cannot resolve external\
-          \ dependency junit:junit:4.12 because no repositories are defined."
-        severity: "ERROR"
-    - id: "Unknown:org.apache.commons:commons-text:1.1"
-      issues:
-      - timestamp: "1970-01-01T00:00:00Z"
-        source: "Gradle"
-        message: "Unresolved: ModuleVersionNotFoundException: Cannot resolve external\
-          \ dependency org.apache.commons:commons-text:1.1 because no repositories\
-          \ are defined."
-        severity: "ERROR"
+    dependencies: []
 packages: []
+issues:
```

Comment:
The GI reports these on the project level instead of per-scope, which probably makes sense as repositories are also defined on the project level. This implies some nice de-duplication.

Reply:
I would say dependencies are defined on the scope level. What if the errors occur further down in the tree, would they also be reporter on the project level instead of inside the tree?

Reply:
> I would say dependencies are defined on the scope level.

True. What I meant was: *Repositories* are defined on the project level, so errors caused by no repository being present could be regarded as project configuration errors, and in that sense it makes sense (heh) to report the errors on the project level. However, missing repos are probably not the only cause for a dependency to be unresolved.

Probably the question boils down to: Is the position in the dependency tree important, i.e. may it have any influence on the error. I guess not.

But I'm undecided. I'm fine either way, to record unresolved dependencies (deduplicated) on the project level, or inside the dependency tree (with possible duplicates).

Reply:
> Probably the question boils down to: Is the position in the dependency tree important, i.e. may it have any influence on the error. I guess not.
> But I'm undecided. I'm fine either way, to record unresolved dependencies (deduplicated) on the project level, or inside the dependency tree (with possible duplicates).

For direct dependencies this might not be so important, but for transitive dependencies it can be very helpful to see where in the tree the issue occurs. I would appreciate if we would report issues inside the tree where possible.

Reply:
Looks like you're still fine for now with the current state as you've approved, so I'll eventually fix this up later to get the tests in now.